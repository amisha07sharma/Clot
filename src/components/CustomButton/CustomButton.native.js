import React from "react";
import { TouchableOpacity } from "react-native";
import PropTypes from "prop-types";
import { Wrapper, TextWrapper } from "./CustomButton.style";

const CustomButton = ({ title, disabled = false, onPress }) => {
  return (
    <TouchableOpacity disabled={disabled} onPress={onPress} testID="Press">
      <Wrapper disabled={disabled}>
        <TextWrapper disabled={disabled}>{title}</TextWrapper>
      </Wrapper>
    </TouchableOpacity>
  );
};

CustomButton.prototype = {
  title: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  onPress: PropTypes.func.isRequired,
};
export default CustomButton;
