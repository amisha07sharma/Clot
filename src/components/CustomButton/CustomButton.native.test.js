import React from "react";
import { render, fireEvent, screen } from "@testing-library/react-native";
import CustomButton from "./CustomButton";

const mockOnPress = jest.fn();
const mockProps = {
  title: "Title",
  onPress: mockOnPress,
};

const renderComponent = (props) => {
  render(<CustomButton {...mockProps} {...props} />);
};
describe("CustomButton component", () => {
  it("renders correctly with default props", () => {
    const wrapper = renderComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly when isActive is true", () => {
    const wrapper = renderComponent({ disabled: true });
    expect(wrapper).toMatchSnapshot();
  });

  it("handles onPress event correctly", () => {
    renderComponent();
    fireEvent(screen.getByTestId("Press"), "onPress");
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });
});
