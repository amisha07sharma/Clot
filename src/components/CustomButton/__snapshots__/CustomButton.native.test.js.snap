// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton component renders correctly when isActive is true 1`] = `undefined`;

exports[`CustomButton component renders correctly with default props 1`] = `undefined`;
